os/drivers/testcase/testcase_drv.c

```diff
@@ -72,8 +78,95 @@ static int testdrv_ioctl(FAR struct file *filep, int cmd, unsigned long arg)
 	 *   ioctl argument:  An integer value indicating the particular test to be run
 	 */
 
-	case TESTIOC_DRIVER_ANALOG: {
+	case TESTIOC_ANALOG: {
+
+	}
+	break;
+
+	case TESTIOC_CLOCK_ABSTIME2TICKS: {
+		base_time.tv_sec = (time_t)arg;
+		base_time.tv_nsec = 0;
+		if (clock_abstime2ticks(CLOCK_REALTIME, &base_time, &ret) != OK) {
+			ret = ERROR;
+		}
+	}
+	break;
+
+	case TESTIOC_GET_SELF_PID: {
+		tcb =  sched_self();
+		if (tcb == NULL) {
+			ret = ERROR;
+			break;
+		}
+		ret = tcb->pid;
+	}
+	break;
+
+	case TESTIOC_GET_SIG_FINDACTION_ADD: {
+		ret = (int)sig_findaction(sched_self(), 2);
+		if (ret == 0) {
+			ret = ERROR;
+			break;
+		}
+	}
+	break;
+
+	case TESTIOC_IS_ALIVE_THREAD: {
+		tcb = sched_gettcb((pid_t)arg);
+		if (tcb == NULL) {
+			ret = ERROR;
+			break;
+		}
+		ret = OK;
+	}
+	break;
+
+	case TESTIOC_GET_TCB_SIGPROCMASK: {
+		tcb = sched_gettcb((pid_t)arg);
+		if (tcb == NULL) {
+			ret = ERROR;
+			break;
+		}
+		ret = tcb->sigprocmask;
+	}
+	break;
+
+	case TESTIOC_GET_TCB_ADJ_STACK_SIZE: {
+		tcb = sched_gettcb((pid_t)arg);
+		if (tcb == NULL) {
+			ret = ERROR;
+			break;
+		}
+		ret = tcb->adj_stack_size;
+	}
+	break;
+
+	case TESTIOC_GET_TCB_TIMESLICE: {
+		tcb = sched_gettcb((pid_t)arg);
+		if (tcb == NULL) {
+			ret = ERROR;
+			break;
+		}
+		ret = tcb->timeslice;
+	}
+	break;
+
+	case TESTIOC_SCHED_FOREACH: {
+		sched_foreach((void *)arg, NULL);
+	}
+	break;
+
+	case TESTIOC_SIGNAL_PAUSE: {
+		ret = pause();
+		if (ret == ERROR) {
+			ret = get_errno();
```

Comment:
Most of them returns OK or ERROR.
why does it return the errno?
If you want to consider EINTR, you need to check it here, not TC I think.